stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/ChatApi.kt

```diff
@@ -181,7 +182,10 @@ internal class ChatApi(
         return retrofitApi.searchMessages(apiKey, connectionId, request)
             .map {
                 it.results.map { resp ->
-                    resp.message
+                    resp.message.apply {
+                        (cid.takeUnless { it.isBlank() } ?: channelInfo?.cid)
```

Comment:
`cid.takeUnless` seems dangerous here. If there isn't a `cid` in the response, this will be `null` instead of being an empty string (thanks to Gson).

Reply:
Oh wait, is this circumvented by `CustomObjectGsonAdapter`'s `null` restoration?

Reply:
If it is not present it will be empty string, because it is initialized by constructor by default parameters.
Gson only set null if literal null is in the json `{ cid: null }`, something that we are not going to handle

Reply:
The default parameters in the ctor don't take effect if the instance is created by Gson, because it's created by a `newInstance` call, and the ctor is never executed. But I believe our custom parser changes the `null` value to an empty string, according to [this impl](https://github.com/GetStream/stream-chat-android/blob/8870fb4d77a63e1c15592c068795af1cff93614d/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/adapters/CustomObjectGsonAdapter.kt#L112-L138) and [this test](https://github.com/GetStream/stream-chat-android/blob/8870fb4d77a63e1c15592c068795af1cff93614d/stream-chat-android-client/src/test/java/io/getstream/chat/android/client/parser/CustomObjectParsingTests.kt#L33-L52).